utils/mk-squash-iso.sh

```diff
@@ -0,0 +1,169 @@
+	to load the squashfs with the following kernel options:
+
+		boot=LABEL="FOOBAR" root=/boot/squashfs.img rootfsflags=loop
+
+	Replace FOOBAR with the value you also specified with -l
+EOF
+}
+
+warn() {
+	echo "WARN: $@"
+}
+
+fail() {
+	echo "FAIL: $@"
+	rm -rf $ISOROOT
+	rm -rf $FAKEROOT
+	exit 1
+}
+
+EXCLUDE="src/build src/log"
+DRY="false"
+
+while [ -n "$1" ]; do
+	case "$1" in
+		'-o')
+			OUTFILE=$2
+			shift 2
+			;;
+		'-l')
+			DISKID=$2
+			shift 2
+			;;
+		'-r')
+			R=$2
+			shift 2
+			;;
+		'-t')
+			EXCLUDE="$EXCLUDE src/tarballs"
+			shift 1
+			;;
+		'-d')
+			DRY="true"
+			shift 1
+			;;
+	esac
+done
+
+[ -z "$R" ] && [ -f config ] && . ./config
+[ -d "$R" ] || fail "$R does not exist or not a directory"
+[ -z "$DISKID" ] && DISKID="sabotage-$ARCH-`date +%Y%m%d`"
+[ -z "$OUTFILE" ] && OUTFILE="$DISKID.iso"
+
+# Check for prerequisite commands
+type mksquashfs >/dev/null \
+	|| fail "mksquashfs not found, try butch install squashfs-tools"
+
+type mkisofs >/dev/null \
+	|| fail "mkisofs not found, try butch install xorriso"
+
+# The tempdir that will become the ISO root and the new /boot
+ISOROOT=$(mktemp -d)
+cp -r $R/boot/* $ISOROOT
+
+# Create directory stubs for the squashfs
+# Those are required as mountpoints for their actual filesystems
+FAKEROOT=$(mktemp -d)
+mkdir $FAKEROOT/boot $FAKEROOT/proc $FAKEROOT/sys $FAKEROOT/dev $FAKEROOT/tmp $FAKEROOT/mnt
+
+if ! [ -f "$ISOROOT/isolinux.cfg" ]; then 
+	warn "/boot/isolinux.cfg not found, generating default config"
+	cat <<EOF > $ISOROOT/isolinux.cfg
+PROMPT 1
+TIMEOUT 100
+DEFAULT sabotage
+
+LABEL ro
+		MENU LABEL Boot with read-only rootfs
+        KERNEL vmlinuz
+		INITRD initrd.img
+        APPEND boot=LABEL="$DISKID" root=/boot/squashfs.img rootfsflags=loop
+
+LABEL sabotage
+		MENU LABEL Boot with RAM overlayfs
+        KERNEL vmlinuz
+		INITRD initrd.img
+        APPEND boot=LABEL="$DISKID" root=/boot/squashfs.img rootfsflags=loop overlayfstype=tmpfs
+
+LABEL rescue
+		MENU LABEL Boot to initramfs
+        KERNEL vmlinuz
+		INITRD initrd.img
+        APPEND boot=LABEL="$DISKID" root=/boot/squashfs.img rootfsflags=loop rescue
+EOF
+fi
+
+# Do some sort of validation
+cat $ISOROOT/isolinux.cfg|grep '^\s*\(KERNEL\|INITRD\)'|awk '{print($2)}' \
+	|sort -u|while read file; do
+	[ -f "$ISOROOT/$file" ] \
+		|| warn "/boot/$file not found, but referenced by isolinux.cfg"
+done
+
+# At least some config has to be available
+[ -f "$ISOROOT/isolinux.cfg" ] \
+	|| fail 'isolinux.cfg not present, iso will not be bootable'
+
+# Check for isolinux.bin ready
+[ -f "$ISOROOT/isolinux.bin" ] || fail "isolinux.bin required, but not found"
+
+# Detect syslinux version
+isoversion=$(strings "$ISOROOT/isolinux.bin"|grep ISOLINUX)
+[ -z "$isoversion" ] && fail "ISOLINUX version could not be detected"
+echo "INFO: $isoversion detected" 
+
+# Check for ldlinux.c32 ready if required
+if echo $isoversion|awk '{if($2<5) exit 1;}'; then
```

Comment:
here is hardcoded a lot of insider knowlegde about specific syslinux versions. would be nice to have a few words about the details either as comments or in the commit message. for example which version is minimum, and which differences/caveats there are between versions.